test/integration/scheduler/queue_test.go

```diff
@@ -336,6 +371,136 @@ func TestCoreResourceEnqueue(t *testing.T) {
 	}
 }
 
+// TestCoreResourceNoEnqueue verify Pods failed by in-tree default plugins
+// will not be moved to active queue by irrelevant events.
+func TestCoreResourceNoEnqueue(t *testing.T) {
```

Comment:
Sorry, bringing [this](https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1597897799) back.

I'm still not quite sure whether we should separate it from TestCoreResourceEnqueue.

> For TestCoreResourceEnqueue, we can use the same logic with QHint enabled/disabled. Because any event that makes a pod schedulable with QHint disabled can always make a pod schedulable with QHint enabled.
However, for TestCoreResourceNOEnqueue. A event that makes a pod schedulable with QHint disabled may can't make a pod schedulable with QHint enabled. So we can't run the same logic with QHint enabled/disabled twice.

Then we can prepare the bool field in each field whether a test case is valid for which value on QHint feature gate.

```go
tests := []struct {
  //...
  featureGates []bool
}
```

> as each plugin adds its logic to the tests, the testing logic can become more complex. I prefer writing two separate tests to make the logic of each test more maintainable.

Yes, this is true. TestCoreResourceEnqueue (for now) is made general and intended to include any plugins' test and hence it could grow too big in the future.
But, `TestCoreResourceNOEnqueue` looks not the right answer. This problem would still exist even after we split the test into TestCoreResourceEnqueue and TestCoreResourceNOEnqueue; Each test would include any plugins' test and hence they could grow too big in the future.

Reply:
Before https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1615027475 . We have different implementations in two tests. So I think it's necessary to separate it to keep the main logic as simple as we can.

But with https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1615027475 implemented. Yes, I think we can reuse TestCoreResourceEnqueue and keep the main test logic remain unchanged.